elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/mapping/CaseMappingGenerator.java

```diff
@@ -76,4 +81,37 @@ private void genericDataMapping(JsonWriter jw, CaseTypeEntity caseType,
         jw.endObject();
         jw.endObject();
     }
+
+    private void aliasMapping(JsonWriter jw, CaseTypeEntity caseType) throws IOException {
+        if (!caseType.getSearchAliasFields().isEmpty()) {
+            log.info("generating search alias field mapping");
+            jw.name(ALIAS);
+            jw.beginObject();
+            jw.name(PROPERTIES);
+            jw.beginObject();
+            for (SearchAliasFieldEntity searchAliasField : caseType.getSearchAliasFields()) {
+                jw.name(searchAliasField.getReference());
+                String aliasMapping = config.getElasticMappings().get(ALIAS).replace(ALIAS_CASE_FIELD_PATH_PLACE_HOLDER, searchAliasField.getCaseFieldPath());
+                jw.jsonValue(aliasMapping);
+                log.info("property: {}, alias mapping: {}", searchAliasField.getReference(), aliasMapping);
+
+                aliasTextSortMapping(jw, searchAliasField);
```

Comment:
shall we rename
`addAliasForTextFieldSorting`

Reply:
done